pkg/apis/autoscaling/types.go

```diff
@@ -420,3 +420,201 @@ type HorizontalPodAutoscalerList struct {
+type PodResourcePolicy struct {
+	// Per-container resource policies.
+	// +optional
+	ContainerPolicies []ContainerResourcePolicy
+}
+
+// ContainerResourcePolicy controls how autoscaler computes the recommended
+// resources for a specific container.
+type ContainerResourcePolicy struct {
+	// Name of the container or DefaultContainerResourcePolicy, in which
+	// case the policy is used by the containers that don't have their own
+	// policy specified.
+	Name string
+	// Whether autoscaler is enabled for the container. Defaults to "On".
+	// +optional
+	Mode ContainerScalingMode
+	// Specifies the minimal amount of resources that will be recommended
+	// for the container.
+	// +optional
+	MinAllowed api.ResourceList
+	// Specifies the maximum amount of resources that will be recommended
+	// for the container.
+	// +optional
+	MaxAllowed api.ResourceList
+}
+
+const (
+	// DefaultContainerResourcePolicy can be passed as
+	// ContainerResourcePolicy.Name to specify the default policy.
+	DefaultContainerResourcePolicy = "*"
+)
+
+// ContainerScalingMode controls whether autoscaler is enabled for a specific
+// container.
+type ContainerScalingMode string
+
+const (
+	// ContainerScalingModeOn means autoscaling is enabled for a container.
+	ContainerScalingModeOn ContainerScalingMode = "On"
+	// ContainerScalingModeOff means autoscaling is disabled for a container.
+	ContainerScalingModeOff ContainerScalingMode = "Off"
+)
+
+// VerticalPodAutoscalerStatus describes the runtime state of the autoscaler.
+type VerticalPodAutoscalerStatus struct {
+	// The most recently computed amount of resources recommended by the
+	// autoscaler for the controlled pods.
+	// +optional
+	Recommendation RecommendedPodResources
+
+	// Conditions is the set of conditions required for this autoscaler to scale its target,
+	// and indicates whether or not those conditions are met.
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []VerticalPodAutoscalerCondition
+}
+
+// RecommendedPodResources is the recommendation of resources computed by
+// autoscaler. It contains a recommendation for each container in the pod
+// (except for those with ContainerScalingMode set to Off).
+type RecommendedPodResources struct {
+	// Resources recommended by the autoscaler for each container.
+	// +optional
+	ContainerRecommendations []RecommendedContainerResources
+}
+
+// RecommendedContainerResources is the recommendation of resources computed by
+// autoscaler for a specific container. Respects the container resource policy
+// if present in the spec.
+type RecommendedContainerResources struct {
+	// Name of the container.
+	Name string
+	// Recommended amount of resources.
+	Target api.ResourceList
+	// Minimum recommended amount of resources.
+	// Running the application with less resources is likely to have
+	// significant impact on performance/availability.
+	// +optional
+	MinRecommended api.ResourceList
```

Comment:
I don't think you need "recommended" in the name here - the block is already named that way.  And Target is not using that suffix either.  Same for max.

Reply:
Renamed to LowerBound / UpperBound.
It's important that it's not confused with MinAllowed / MaxAllowed from the spec, so I prefer these names to not be too similar.